code/modules/mob/living/simple_animal/hostile/megafauna/hierophant.dm

```diff
@@ -108,10 +108,9 @@ Difficulty: Hard
 	else
 		stat = DEAD
 		blinking = TRUE //we do a fancy animation, release a huge burst(), and leave our staff.
-		burst_range = 10
 		visible_message("<span class='hierophant'>\"Mrmxmexmrk wipj-hiwxvygx wiuyirgi...\"</span>")
 		visible_message("<span class='hierophant_warning'>[src] shrinks, releasing a massive burst of energy!</span>")
-		burst(get_turf(src))
+		INVOKE_ASYNC(GLOBAL_PROC, .proc/hierophant_burst, null, get_turf(src), 10)
```

Comment:
Why change the caster?

Reply:
In fact why not just `INVOKE_ASYNC` the `burst()` proc

Reply:
Because the hierophant's just about to die anyway, so there's no point in setting it as the caster.

Reply:
Didn't see the second message there, asking further on irc

Reply:
The reference seems to suggest that deleting an object will silently kill any processes it's invoking, which would be pretty bad.